bindings/pyroot/pythonizations/python/ROOT/_pythonization/_function_utils.py

```diff
@@ -0,0 +1,224 @@
+            else: # the parameter was not in funcargs. Thus this parameter has to be mapped to a column of rdf
+                if p not in self.col_names:
+                    raise Exception(f"Unable to map function argument {p} to a column.\nUse correct name of column or use funcargs.")
+                input_args[p] = p 
+                type_args[p] = self.find_type(p)
+        return input_args, type_args
+
+
+    def generate_function_call(self, func, funcargs):
+        """
+        Function to generate the c++ function call from the Numba Namespace.
+        Also gives a name to the function if the function is a lambda function.
+        Args:
+	    `   func: Function whose signature is to be determined.
+                    P.S The function should not be crazy. It should have simple datatypes for the arguments and should not have default arguments. (Use funcargs instead)
+            funcargs: Dictionatry containing constant parameters the that func requires. Or any special mapping of the arguments of func.
+
+        Returns:
+            func_call: String of function's c++ call which can be passed to the corresponding define or filter.
+            func_sign: The signature of the function which is needed for jitting.
+        """
+        input_args, type_args = self.find_function_signature(func, funcargs)
+        if func.__name__ == '<lambda>' : 
+            try:
+                from uuid import uuid4
+            except:
+                raise ImportError("Could not import uuid module to give unique identifier to lambda functions.")
+            func.__name__ = "func_"+uuid4().__str__().split('-')[-1]
+        func_call = func.__name__ + "("
+        func_sign = []
+        for p in self.params:
+            func_call = func_call + str(input_args[p]) + ", "
+            func_sign.append(str(type_args[p]))
+        if len(self.params) != 0:
+            func_call = func_call[:-2] + ")"
+        else:
+            func_call = func_call + ")"
+        return func_call, func_sign
+            
+    def jit_function(self, func, funcargs):
+        """
+        Function to jit the function
+        Args:
+	    `   func: Function whose signature is to be determined.
+                    P.S The function should not be crazy. It should have simple datatypes for the arguments and should not have default arguments. (Use funcargs instead)
+            funcargs: Dictionatry containing constant parameters the that func requires. Or any special mapping of the arguments of func.
+        """
+        if func.__name__ in FunctionSignature.function_cache.keys():
+            func_call, func_sign = FunctionSignature.function_cache[func.__name__]
+            try:
+                import warnings 
+            except:
+                raise ImportError("Cuuld not import warnings to flag warning suring function jitting.")
+            warnings.warn(f"Trying to re-jit function: {func.__name__}.\n Do not change function signature.")
```

Comment:
I don't understand this warning, why are we re-jitting if the function was already in the cache?

Reply:
Its a warning telling the user that you are tying to re-jit the function.
Since this is not possible. (Can't have two functions with same name in the namespace)
I intended to warn the user that, use the function but don't change its signature. (don't pass an array instead of an int)

Reply:
Changed to an error if the signature does not match but the names do.